### ISSUE TITLE
Prophet Seems to standardize regressors even when `standardize = False` ?

### ISSUE THREAD

**Issue by janethompson** [facebook/prophet#1190]
Hi there,

I am running the prophet model with some regressors and I noticed that prophet seems to be standardizing the values even when standardize is set to false. Some sample data that I am processing looks like this:

![image](https://user-images.githubusercontent.com/54864102/67489884-4ce61e00-f640-11e9-9825-afc353a1640e.png)

I defined my own function in python to forecast my data set. After running my data through my funciton, the output of this is looks like this, with a duplicate of each regressor that seems to be standardized again:

![image](https://user-images.githubusercontent.com/54864102/67509816-d22dfa80-f661-11e9-85b0-ead28f15ce40.png)

**Comment by janethompson** [comment]
The function call looks like this:
```states = ['AK','AL','AR','AZ','CA','CO','CT','DE','FL','GA','HI','IA','ID','IL','IN','KS','KY','LA','MA','MD','ME','MI','MN','MO','MS','MT','NC','ND','NE','NH','NJ','NM','NV','NY','OH','OK','OR','PA','RI','SC','SD','TN','TX','UT','VA','VT','WA','WI','WV','WY']
socs = ['RNs', 'Stat', 'DentHyg', 'ManageAn', 'SecAdmin', 'DataArch']
demand_x = regressors[['ds', 'STATE', 'SOC3_MAP', 'APPLY_START']]

rf_dem = linear_prophet(state_list = states, 
                                 soc_list = socs, 
                                 dat_train = dem_tars[dem_tars.ds <= '2019-08-31'], 
                                 dat_test = dem_tars[dem_tars.ds <= '2019-09-30'],
                                    regressor_data = demand_x)```

And the function definition is:

``def linear_prophet(state_list, soc_list, dat_train, dat_test, regressor_data = None, strict_regressors = None):    
    results = pd.DataFrame(columns=['ds','yhat','y','state_soc'])
    skipped = []
    expected_train_length = dat_train.ds.nunique()
    for state in tqdm(state_list):
        for soc in tqdm(soc_list):
            
            # subset the train and test data to only the state/soc combination in question
            train_chunk = dat_train[(dat_train.STATE == state) & (dat_train.SOC3_MAP == soc)].copy()
            test_chunk = dat_test[(dat_test.STATE == state) & (dat_test.SOC3_MAP == soc)]
            
            # if the combination does not exist, return to next combination
            if (train_chunk.shape[0] < expected_train_length):
                print(f'Skipping iteration for {soc} in {state}, not enough training data')
                skipped.append(state+'_'+soc)
                continue
            if  (test_chunk.shape[0] == 0):
                print(f'Skipping iteration for {soc} in {state}, no test data available')
                skipped.append(state+'_'+soc)
                continue
            
            # instantiate a prophet object. With growth = logistic 0 is the implicit minimum and we must specify a cap for each row
            
            # 16 prophet parameters
            model = Prophet(growth='linear',             # linear or logistic
                            changepoints=None,           # let this be decided automatically
                            n_changepoints=4,            # allow between 0 and 12 changepoints
                            changepoint_prior_scale=0.05, # flexibility of automatic changepoints selection
                            changepoint_range=0.8,       # proportion of history for which changepoints will be calculated
                            yearly_seasonality=False, 
                            weekly_seasonality=False, 
                            daily_seasonality=False, 
                            holidays=None, 
                            seasonality_mode='additive',  # additive or multiplicative
                            seasonality_prior_scale=10.0, # smaller values dampen the seasonality
                            holidays_prior_scale=1,    # default is 10
                            mcmc_samples=0, 
                            interval_width=0.8, 
                            uncertainty_samples=10)
            
            model.add_seasonality(name = 'monthly', 
                                  period = 30.5, 
                                  fourier_order = 6, # yearly is 10, weekly is 3; higher values risk overfitting
                                  prior_scale = 10,
                                  mode = 'additive')
#             model.add_seasonality(name = 'quarterly',
#                                  period = 365.25/4,
#                                  fourier_order = 8,
#                                  prior_scale = 10)
            
            # add regressor data if regressor_data is passed and reset train and test chunk datasets
            if regressor_data is not None:
            
                strict_regressors = [x for x in list(regressor_data.columns) if x not in ['ds', 'STATE', 'SOC3_MAP']]
                regressor_chunk = regressor_data[(regressor_data.STATE == state) & (regressor_data.SOC3_MAP == soc)]
                train_chunk = train_chunk.merge(regressor_chunk, how = 'inner', on = ['ds', 'STATE', 'SOC3_MAP'])
                test_chunk = test_chunk.merge(regressor_chunk, how = 'inner', on = ['ds', 'STATE', 'SOC3_MAP'])
            
            
                # if regressor data is available, add the regressors to the model
                for regressor in strict_regressors:
                    # should i use additive or multiplicative here?
                    # we have already pre-scaled our data using robustscaler()
                    model.add_regressor(regressor, prior_scale=0.5, mode='additive', standardize = False)


            
            
            # fit the model to the training data 
            model.fit(train_chunk)
            future_data = model.make_future_dataframe(periods=1, freq = 'm')
            
            # define merge fields to include regressors, if present
            if strict_regressors is None:
                merge_fields = ['ds'] #['ds', 'cap']
            else:
                merge_fields = strict_regressors + ['ds'] #['ds', 'cap']
            
            # add regressors to future data if regressor variables are passed:
            if regressor_data is not None:
                # test_chunk should be the entire dataset, not just the one row (it is fit to the training data and then used to
                # predict on the unseen row
                future_data = future_data.merge(test_chunk[merge_fields], how = 'inner', on = ['ds'])  
                forecast_data = model.predict(future_data)[['ds', 'yhat']+strict_regressors]
            
            # if no regressors are present make predictions using just time
            else:
                future_data = future_data.merge(test_chunk[merge_fields], how = 'inner', on = ['ds'])  
                forecast_data = model.predict(future_data)[['ds', 'yhat']]

            
            # combine actuals with predictions and store the result
            result = forecast_data.merge(test_chunk, how = 'inner', on = 'ds')
            
        
            # save the result to location "results"
            results = pd.concat([results, result], axis = 0, sort = True)

    return results, skipped```

**Comment by janethompson** [comment]
The clean_predictions functions is just a little processing of negative predictions or places where the actual was equal to 0:

```def clean_predictions(data, test_date):
    
    # convert actual (y) to type float
    data.y = data.y.astype(float)
    
    # handled negative and 0 values to allow for smooth MAE calculations
    data['cleaned_y'] = np.where(data.y == 0, .00001, data.y)
    data['cleaned_yhat'] = np.where(data.yhat < 0, data.yhat*(-1), data.yhat)
    
    # calculate MAE and MAPE
    data['MAPE'] = ((data.cleaned_y - data.cleaned_yhat) / data.cleaned_y).abs()
    data['MAE'] = (data.cleaned_y - data.cleaned_yhat).abs()
    
    # certain MAPE values will be on the order of a million (where actual ~0, predicted = 15)
    # this will hugely skew our distribution of error so for now let's cheat and force these high values down to 3
    data.MAPE = np.where(data.MAPE > 3, 3, data.MAPE)
    data.MAPE = np.where((data.MAE < 5) & (data.MAPE > .2), .19, data.MAPE)
    
    # define test and train chunks
    data['test_set'] = np.where(data.ds == test_date, 1, 0)
    
    return data```

**Comment by bletham** [comment]
OK, I think I see what is happening here.

The `linear_prophet` function is producing a forecast, and then extracting ds, yhat, and the columns for the regressors:
```
forecast_data = model.predict(future_data)[['ds', 'yhat']+strict_regressors]
```
Then, this is merged with the test data for the purpose of computing error metrics:
```
result = forecast_data.merge(test_chunk, how = 'inner', on = 'ds')
```
Here, `test_chunk` and `forecast_data` both have columns for the regressors with the same name (APPLY_START in the screenshot above), so pd.merge is renaming one to have _x and the other to have _y. APPLY_START_x is the column from `forecast_data` and APPLY_START_y is that from `test_chunk`.

As discussed in #1188, the value in `forecast_data` is not the input to the model but rather the output: it is how much of `y` is being attributed to APPLY_START. If you call `m.plot_components` to look at the components plot, it is the values shown in the plot.

In this case since standardization was turned off, the contribution to the model is exactly `beta * APPLY_START` where `beta` is a parameter fit to the data. You can see that `beta=0.138756` which is what produces the values in `forecast_data`.

One thing that I will note is that prophet assumes a continuous y and so might not perform well on data where y is discrete and very small (like the 1-10 seen in the figure above); this is something we're working on improving but in the meantime if there's a meaningful way that data can be aggregated that could improve predictions significantly.

**Comment by janethompson** [comment]
Ahhh, that makes sense! Thanks for taking a look Ben. If you were to recommend a threshold, how large would you say y needs to be in order to get a meaningful forecast using prophet? (ok if that's not easy to answer). Just trying to understand if we aggregate up to say 25-50 would that be good enough or were you thinking more in the ballpark of 100-200 (or perhaps up over 1000) ?

**Comment by bletham** [comment]
Sorry for the extremely long delay in my reply - my expectation is that aggregation around 25-50 would be sufficient.
